IHP/HtmlSupport/Parser.hs

```diff
@@ -126,13 +146,16 @@ hsxClosingElement name = do
 
 hsxChild = hsxElement <|> hsxSplicedNode <|> hsxText
 
+-- | Parses a hsx text node
+--
+-- Stops parsing when hitting a variable, like `{myVar}`
 hsxText :: Parser Node
-hsxText = do
-    value <- takeWhile1P (Just "text") (\c -> c /= '{' && c /= '}' && c /= '<' && c /= '>')
-    let isWhitespaceTextOnly = Text.null (Text.strip value)
-    pure if isWhitespaceTextOnly
-        then TextNode ""
-        else TextNode (collapseSpace value)
+hsxText = buildTextNode <$> takeWhile1P (Just "text") (\c -> c /= '{' && c /= '}' && c /= '<' && c /= '>')
+
+-- | Builds a TextNode and strips all surround whitespace from the input string
+buildTextNode :: Text -> Node
+buildTextNode value | Text.null (Text.strip value) = TextNode ""
+buildTextNode value = TextNode (collapseSpace value)
```

Comment:
Bit of a nit-pick, but I find it easier to parse the logic here if you just stick to guards:

```suggestion
buildTextNode value 
    | Text.null (Text.strip value) = TextNode ""
    | otherwise = TextNode (collapseSpace value)
```

Reply:
Agree :) Will change this